lib/bolt_spec/plans/action_stubs/plan_stub.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module BoltSpec
+  module Plans
+    class PlanStub < ActionStub
+      def matches(_scope, _plan, params)
+        targets = params.fetch(:nodes, params.fetch(:targets, nil))
+        if @invocation[:targets] && Set.new(@invocation[:targets]) != Set.new(targets)
+          return false
+        end
+
+        if @invocation[:params] && params != @invocation[:params]
+          return false
+        end
+
+        true
+      end
+
+      def call(_scope, plan, params)
+        @calls += 1
+        if @return_block
+          check_plan_result(@return_block.call(plan: plan, params: params), plan)
```

Comment:
`check_plan_result` isn't defined anywhere.

More generally, the `ActionStub` base class is pretty specific to an action that gets performed on a set of targets and returns a `ResultSet`. It may be more trouble than it's worth to try to adapt that to also work for plans, so we might want to just implement `PlanStub` on its own without inheriting behavior from `ActionStub`.

Reply:
`check_plan_result` is now defined in action_stubs.rb